modules/imgcodecs/src/grfmt_jpeg2000.cpp

```diff
@@ -476,6 +477,18 @@ bool  Jpeg2KEncoder::write( const Mat& _img, const std::vector<int>& )
     if( channels > 3 || channels < 1 )
         return false;
 
+    CV_Assert(params.size() % 2 == 0);
+    double target_compression_rate = 1.0;
+    for( size_t i = 0; i < params.size(); i += 2 )
+    {
+        switch(params[i])
+        {
+        case cv::IMWRITE_JPEG2000_JASPER_COMPRESSION:
+            target_compression_rate /= std::max(params[i+1], 1);
```

Comment:
It seems that this haven't to behave well:
- OpenCV parameter value has "int" type
- valid parameter values are 1,2,3,4,...
- corresponding `target_compression_rate` are: 1.0, 0.5, 0.333, 0.25

As we can see target values set is very discreet.
Do we really need it in this way? (for example, we don't need 66%, 80% compression)

Reply:
You are right that the values are discrete due to the `int` type.
If you prefer the possible values to be equidistant, I could change the implementation to something similar to the JPEG quality parameter (0-100), where one integer means 1% and 0 corresponds to the smallest possible compression rate.
In this case one could rename `COMPRESSION` to `QUALITY`.
What do you think?

Reply:
Looks like JPEG2000 codecs may have both parameters "COMPRESSION" and "QUALITY":
- OpenJPEG compression tool: https://github.com/uclouvain/openjpeg/wiki/DocJ2KCodec (`-r` and `-q`)
- Jasper compression: https://www.ece.uvic.ca/~frodo/jasper/jasper.pdf (page 17 - example 3,4 for compression)
- Jasper quality - ?

and they are different (by behavior).

What is about these names:
- `IMWRITE_JPEG2000_QUALITY` - range 0-100 (if Jasper has this parameter)
- `IMWRITE_JPEG2000_COMPRESSION_X1000`, parameter value is multiplied by 1e-3

Reply:
Thanks for your further investigation!
I scanned again the JasPer documentation and it seems that there's no quality parameter.
Therefore, my proposal would be the following:
 - introduce the parameter `IMWRITE_JPEG2000_COMPRESSION_X1000` proposed by you
 - do not introduce `IMWRITE_JPEG2000_QUALITY` now

Once JasPer is removed in favour of OpenJPEG, one can maintain `IMWRITE_JPEG2000_COMPRESSION_X1000` with the same meaning as with JasPer and add a new parameter `IMWRITE_JPEG2000_QUALITY` which would have a similar meaning as the JPEG quality parameter.
I believe this procedure would be the most future-proof version discussed so far.
What do you think?

Reply:
Agreed!